donkeycar/parts/hall_sensor_rpm.py

```diff
@@ -0,0 +1,58 @@
+"""
+Hall sensor RPM
+"""
+
+#!/usr/bin/python3
+from time import sleep
+import time, math
+
+class HallSensorRpm:
+    def __init__(self, sensor, wheel_radius):
+        import RPi.GPIO as GPIO
+        GPIO.setmode(GPIO.BCM)
+        GPIO.setwarnings(False)
+        GPIO.setup(sensor,GPIO.IN,GPIO.PUD_UP)
+        GPIO.add_event_detect(sensor, GPIO.FALLING, callback = self.calculate_elapse, bouncetime = 20)
+
+        self.rpm = 0
+        self.pulse = 0
+        self.start_timer = time.time()
+        self.wheel_radius = wheel_radius
+        self.on = True
+        self.debug = debug
+        self.wheel_radius = wheel_radius
+
+    def calculate_elapse(self, channel):
+        self.pulse+=1
+        self.elapse = time.time() - start_timer
+        self.start_timer = time.time()
+
+    def update(self):
+        while (self.on):
```

Comment:
I believe update() is called in the main thread. so infinite while loop here would basically hijack the whole main thread..?